src/core/channel/subchannel_call_holder.c

```diff
@@ -168,21 +168,23 @@ void grpc_subchannel_call_holder_perform_op(grpc_exec_ctx *exec_ctx,
 
 static void subchannel_ready(grpc_exec_ctx *exec_ctx, void *arg, bool success) {
   grpc_subchannel_call_holder *holder = arg;
-  grpc_subchannel_call *call;
   gpr_mu_lock(&holder->mu);
   GPR_ASSERT(holder->creation_phase ==
              GRPC_SUBCHANNEL_CALL_HOLDER_PICKING_SUBCHANNEL);
-  call = GET_CALL(holder);
-  GPR_ASSERT(call == NULL || call == CANCELLED_CALL);
   holder->creation_phase = GRPC_SUBCHANNEL_CALL_HOLDER_NOT_CREATING;
   if (holder->connected_subchannel == NULL) {
     fail_locked(exec_ctx, holder);
   } else {
-    gpr_atm_rel_store(
```

Comment:
The answer to this question is that by the tie we reach line 156 we've already checked the value of call and determined that it is definitely NULL. We've already separately handled the CANCELED_CALL case and all other values. Additionally, all modifications to this value are under the lock. Arguably, then, all CAS'es should be gone and replaced with load/store pairs (though we do still need an atomic load for when it is checked outside a lock). That is a TODO for later.